cmake/cpu_extension.cmake

```diff
@@ -0,0 +1,80 @@
+set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
+
+include_directories("${CMAKE_SOURCE_DIR}/csrc")
+
+#
+# Check the compile flags
+#
+list(APPEND CXX_COMPILE_FLAGS 
+    "-fopenmp"
+    "-DVLLM_CPU_EXTENSION")
+
+execute_process(COMMAND cat /proc/cpuinfo
+                RESULT_VARIABLE CPUINFO_RET
+                OUTPUT_VARIABLE CPUINFO)
+
+if (NOT CPUINFO_RET EQUAL 0)
+    message(FATAL_ERROR "Failed to check CPU features via /proc/cpuinfo")
+endif()
+
+function (find_isa CPUINFO TARGET OUT)
+    string(FIND ${CPUINFO} ${TARGET} ISA_FOUND)
+    if(NOT ISA_FOUND EQUAL -1)
+        set(${OUT} ON PARENT_SCOPE)
+    else()
+        set(${OUT} OFF PARENT_SCOPE)
+    endif()
+endfunction()
+
+find_isa(${CPUINFO} "avx512f" AVX512_FOUND)
+
+if (AVX512_FOUND)
+    list(APPEND CXX_COMPILE_FLAGS
+        "-mavx512f"
+        "-mavx512vl"
+        "-mavx512bw"
+        "-mavx512dq")
+
+    find_isa(${CPUINFO} "avx512_bf16" AVX512BF16_FOUND)
+    if (AVX512BF16_FOUND AND
+        CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND 
+        CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 12.3) 
+        list(APPEND CXX_COMPILE_FLAGS "-mavx512bf16")
+    else()
+        message(WARNING "Disable AVX512-BF16 ISA support, requires gcc/g++ >= 12.3")
```

Comment:
This will get triggered even if `AVX512BF16_FOUND` is false. Instead, should we do something like this?

```cmake
if (AVX512BF16_FOUND)
    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 12.3)
        list(APPEND CXX_COMPILE_FLAGS "-mavx512bf16")
    else()
        message(WARNING "AVX512-BF16 ISA supported but disabled, requires gcc/g++ >= 12.3")
    endif()
    message(WARNING "AVX512-BF16 ISA not supported")
endif()

Reply:
Nice catch! Have fixed it with better message.